TaskWithSmartPointers/MySharedPtr.h

```diff
@@ -0,0 +1,113 @@
+#pragma once
+#include <iostream>
+
+template <class T>
+class MySharedPtr
+{
+private:
+	T* m_ptr = nullptr;
+	long* m_referencesCounter = nullptr;
+	friend std::ostream& operator<<(std::ostream& os, MySharedPtr const& m);
+public:
+	T* get()
+	{
+		return m_ptr;
+	}
+
+	long use_count() const noexcept
+	{
+		if (m_referencesCounter != nullptr)
```

Comment:
The short form of this is:
`return m_referencesCounter ? *m_referencesCounter : 0`;

Reply:
Done